src/EditorFeatures/Core/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -237,6 +244,27 @@ public IEnumerable<char> PotentialCommitCharacters
                 return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
             }
 
+            // Specifically for snippets, we use reflection to try and invoke the LanguageServerSnippetExpander's
+            // TryExpand method and determine if it succeeded or not.
+            if (SnippetCompletionItem.IsSnippet(roslynItem))
+            {
+                change.Properties!.TryGetValue("LSPSnippet", out var lspSnippetText);
+
+                if (!document.TryGetText(out var sourceText))
```

Comment:
you have the trigger snapshot and the current snapshot for the buffer.  do we need the sourceText given that?